### ISSUE TITLE
control CNAME config from repo

### ISSUE THREAD

**Issue by morph027** [YuMS/gitlab-ce-pages#14]
If e want to enable users to deploy pages on their own (ok, after the GCP instance is up and running), we should think about adding a static file into the repo which contains the CNAME config instead of doing it on a volume like now.

Let's think of a static file like `.cnames` which contains the same info like `cnames.txt` for each repo.

Then, if the deployer finds such a file, it will kick of the configuration.

**Comment by YuMS** [comment]
Not until I see this issue did I understand your https://github.com/YuMS/gitlab-ce-pages/issues/6#issuecomment-253150650.

Here are my thoughts.

We are building GCP for GitLab CE users, which are mostly small businesses or groups. Their needs are mainly hosting their code as well as publishing corresponding docs or something somewhere. Therefore, I think it more convenient to let the guy who maintains GitLab CE to handle CNAMEs/SSL.

What you are building is more like GitLab Pages: anyone could link an arbitrary domain name to our host. I think this is needed for larger groups or companies, but they will choose to use GitLab EE, won't they?

**Comment by morph027** [comment]
Some do, some (unfortunately) won't. I'm dealing with both of them ;)

One of my customers is having dozens of keen web developers, which picked up the feature very fast. And they always are mailing me to add the cname entry ;)

So if this is out of scope here, i might think of something different.

**Comment by YuMS** [comment]
Fine, I think wildcard CNAME may save you for now. And I'm curious about why web developers are keen on developing **static** web sites.

**Comment by morph027** [comment]
:smile:

Besided their real tasks, they are creating documentations for their projects.
